From dffe2753885c7ee06c4109c326056e648b026a8e Mon Sep 17 00:00:00 2001
From: Allen Xu <b45815@freescale.com>
Date: Wed, 24 Sep 2014 15:01:57 -0500
Subject: [PATCH 0730/5242] ENGR00333049 ARM: dts: add dts file for ecspi on
 imx6q sabreauto

commit  3d070ed74760aa5b64f298c77ad97f2100b1ee09 from
https://source.codeaurora.org/external/imx/linux-imx.git

add new dts file to support ecspi on imx6q sabreauto board.

Signed-off-by: Allen Xu <b45815@freescale.com>
(cherry picked and merge from commit ff87b56022d1d63003eff79a30bedf33271781fa)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                  |    1 +
 arch/arm/boot/dts/imx6q-sabreauto-ecspi.dts |   43 +++++++++++++++++++++++++++
 2 files changed, 44 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6q-sabreauto-ecspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index cc1b34a..37781cc 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -490,6 +490,7 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6q-pistachio.dtb \
 	imx6q-rex-pro.dtb \
 	imx6q-sabreauto.dtb \
+	imx6q-sabreauto-ecspi.dtb \
 	imx6q-sabreauto-gpmi-weim.dtb \
 	imx6q-sabrelite.dtb \
 	imx6q-sabresd.dtb \
diff --git a/arch/arm/boot/dts/imx6q-sabreauto-ecspi.dts b/arch/arm/boot/dts/imx6q-sabreauto-ecspi.dts
new file mode 100644
index 0000000..4fbb207
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sabreauto-ecspi.dts
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6q-sabreauto.dts"
+
+&ecspi1 {
+	pinctrl-assert-gpios = <&gpio5 4 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+/* remove #if 0 after flexcan is ready */
+#if 0
+&flexcan2 {
+	/* max7310_c on i2c3 is gone */
+	status = "disabled";
+};
+#endif
+
+&i2c3 {
+	/* pin conflict with ecspi1 */
+	status = "disabled";
+};
+
+&uart3 {
+	/* the uart3 depends on the i2c3, so disable it too. */
+	status = "disabled";
+};
+
+&usbh1 {
+	/* max7310_b on i2c3 is gone */
+	status = "disabled";
+};
+
+&usbotg {
+	/* max7310_c on i2c3 is gone */
+	status = "okay";
+	dr_mode = "peripheral";
+};
-- 
1.7.9.5

